File: backend/src/main/java/com/meetr/domain/entity/Notification.java
package com.meetr.domain.entity;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Notification {
    private Long id;
    private String userId;
    private String eventType;
    private String title;
    private String content;
    private Long bookingId;
    private Long roomId;
    private Long bookingStartTimeMs;
    private Long bookingEndTimeMs;
    private Boolean isRead;
    private Long readAt;
    private Long createdAtMs;

    public void markAsRead() {
        this.isRead = true;
        this.readAt = System.currentTimeMillis();
    }
}

File: frontend/src/types/notification.ts
export interface Notification {
  id: number
  userId: string
  eventType: string
  title: string
  content: string
  bookingId: number | null
  roomId: number | null
  bookingStartTimeMs: number | null
  bookingEndTimeMs: number | null
  isRead: boolean
  readAt: number | null
  createdAtMs: number
}
